astropy/io/fits/util.py

```diff
@@ -583,10 +583,13 @@ def _array_from_file(infile, dtype, count):
         # their underlying file object, instead of the decompressed bytes
         read_size = np.dtype(dtype).itemsize * count
         s = infile.read(read_size)
-        array = np.frombuffer(s, dtype=dtype, count=count)
-        # copy is needed because np.frombuffer returns a read-only view of the
-        # underlying buffer
-        array = array.copy()
+        if s == b'':
```

Comment:
I am still scarred from the Python 2 to 3 transition. Are you sure this works fine in Python 2 as well, given you milestoned it to 2.x?

Reply:
I think it should work, but hopefully we can get rid of the `if` with Marten's suggestion.